### PR TITLE
Fix Epinio state in Installer

### PR DESCRIPTION
**Pull request by enrichman**
The Epinio state was not handled, causing a misalignment while switching tabs.

Refactored Installer component to functional approach, added smaller scoped functions, added check for Epinio release.

### CODE REVIEW

File: ui/src/epinio/Installer.js [hunk 1]
```diff
@@ -1,175 +1,205 @@
...
+  hasKubernetes,
+  domain,
+  onInstallationChanged,
+  onError
+}) {
+  const [progress, setProgress] = useState(0)
+  const [installed, setInstalled] = useState(null)
+
+  async function helm(args) {
+    const result = await window.ddClient.extension.host.cli.exec('helm', args)
```
[the remark this answers is not in the document]
Comment (enrichman) [reply ui/src/epinio/Installer.js:15]: Thanks, I've implemented the requested changes!  Regarding the last point:

> This is more of a preference, for this kind of scenarios I don't think the performance is an issue but, functions that you create inside the component they'll be re-created with each reader (if the component re-renders, the functions are also created again). Usually, you can move the functions outside (which will create the function once on the first mount) the component and just call them with the given args.

This is true also declaring a function like this `const helm = async (args) => { .. }`? I would say yes, so I'm in favour of moving them out eventually. Does it make sense to extract some of these into an `Helm` file, or package?

**Comment by mmartin24** [comment]
Checked the following on the latest branch:

- `Case 1`: Click Install, check disabled buttons, change focus, and expect installation completed

Buttons get disabled but after checking focus it starts from 0 and installation is not completed. Is it expected?

https://github.com/epinio/extension-docker-desktop/assets/37271841/688c5643-7f46-417c-ba7a-fd5d7fe2fbed


- `Case 2`: With the extension installed, click `Uninstall`, check the disabled buttons, change focus, and expect installation completed 

Works ok

https://github.com/epinio/extension-docker-desktop/assets/37271841/8d3dc6c4-604f-4551-a46e-ac851f1b4982



- `Case 3`: With Extension enabled/disabled, changing focus preserves the desired status 

Works ok

- `Corner case`: Check the status of deployed apps shown on extension remains when changing focus

It works but takes around 10 seconds. During that time the app seems missing. Is this expected?

https://github.com/epinio/extension-docker-desktop/assets/37271841/8380d6d4-805c-4dfb-b0b0-4a122a3bb6ee


- Other checks: 

Checked deploying app from UI wokrs
Checked deploying app from Extension not working all the time.

**Comment by enrichman** [comment]
> Checked the following on the latest branch:
> 
> * `Case 1`: Click Install, check disabled buttons, change focus, and expect installation completed
> 
> Buttons get disabled but after checking focus it starts from 0 and installation is not completed. Is it expected?
>

Yes, it's expected because it will install nginx, cert-manager and epinio in this order. If you change state during one of the previous installation then it will not complete the epinio one (it will probably just complete nginx or nginx+cert-manager.

>  case1.mov 
> * `Case 2`: With the extension installed, click `Uninstall`, check the disabled buttons, change focus, and expect installation completed
> 
> Works ok

:+1: 

>  case2.mov 
> * `Case 3`: With Extension enabled/disabled, changing focus preserves the desired status
> 
> Works ok

:+1: 

> * `Corner case`: Check the status of deployed apps shown on extension remains when changing focus
> 
> It works but takes around 10 seconds. During that time the app seems missing. Is this expected?
> 
>  lengthy_refresh.mov 
> * Other checks:
> 
> Checked deploying app from UI wokrs Checked deploying app from Extension not working all the time.

This fix is only about the installation status, so I would say that we are fine. :slightly_smiling_face: 
Thanks for the checks!

**Comment by mmartin24** [comment]
thanks @enrichman for looking all the comments [above](https://github.com/epinio/extension-docker-desktop/pull/44#issuecomment-1542330882). I guess then this pr should be ready to go.

**Comment by mmartin24** [comment]
FTR: The problem commented [above](https://github.com/epinio/extension-docker-desktop/pull/44#issuecomment-1542137239) about deploying the app extension not working all the time seemed to have been caused by lack of disk space. I checked it again with more space and worked as commented [here](https://github.com/epinio/extension-docker-desktop/issues/41#issuecomment-1543621998).

**Pull request merged**